circuits/types/field/src/compare.rs

```diff
@@ -21,25 +21,63 @@ impl<E: Environment> Compare<Field<E>> for Field<E> {
 
     /// Returns `true` if `self` is less than `other`.
     fn is_less_than(&self, other: &Self) -> Self::Output {
-        let mut is_less_than = Boolean::constant(false);
-        let mut are_previous_bits_equal = Boolean::constant(true);
-
-        // Initialize an iterator over `self` and `other` from MSB to LSB.
-        let self_bits_be = self.to_bits_be();
-        let other_bits_be = other.to_bits_be();
-        let bits_be = self_bits_be.iter().zip_eq(&other_bits_be);
-
-        for (index, (self_bit, other_bit)) in bits_be.enumerate() {
-            // Determine if `self` is less than `other` up to the `index`-th bit.
-            is_less_than |= &are_previous_bits_equal & (!self_bit & other_bit);
-
-            // Skip the update to the LSB, as this boolean is subsequently discarded.
-            if index != self_bits_be.len() - 1 {
-                are_previous_bits_equal &= self_bit.is_equal(other_bit);
-            }
+        if self.is_constant() && other.is_constant() {
```

Comment:
a `match` expression against `(self.is_constant(), other.is_constant())` might be a bit more readable here